src/codeu/chat/client/commandline/Chat.java

```diff
@@ -402,15 +453,28 @@ public void invoke(List<String> args) {
                             message != null;
                             message = message.next()) {
           System.out.println();
-          System.out.format("USER : %s\n", message.message.author);
+          System.out.format("USER : %s\n", findUsername(message.message.author));
           System.out.format("SENT : %s\n", message.message.creation);
           System.out.println();
           System.out.println(message.message.content);
           System.out.println();
         }
         System.out.println("---  end of conversation  ---");
       }
+      
+      // Find the user with the given UUID and return their username.
+      // If no user has the given UUID this will return "Anonymous".
+      private String findUsername(Uuid author) {
+        for (final UserContext user : context.allUsers()) {
```

Comment:
This method shares a lot of code with `findUser` below. It would be great do reduce some of the duplication here by just calling `findUser` and returning "Anonymous" if `findUser` returns null.